barretenberg/cpp/src/barretenberg/ultra_honk/ultra_prover.cpp

```diff
@@ -45,26 +45,25 @@ template <UltraFlavor Flavor> void UltraProver_<Flavor>::execute_preamble_round(
  */
 template <UltraFlavor Flavor> void UltraProver_<Flavor>::execute_wire_commitments_round()
 {
-    // Commit to the first three wire polynomials
-    // We only commit to the fourth wire polynomial after adding memory records
-    auto wire_polys = instance->proving_key->get_wires();
+    instance->compute_wire_commitments();
```

Comment:
Is it appropriate for the instance to be the entity responsible for committing to witnesses? I'm a little confused about the abstraction but maybe I'm missing some context. I'd like to have a clear articulation of what an instance is so I can better understand what the division of labor should be.